Source/RapyutaSimulationPlugins/Public/Drives/RRPhysicsJointComponent.h

```diff
@@ -51,6 +90,87 @@ class RAPYUTASIMULATIONPLUGINS_API URRPhysicsJointComponent : public URRJointCom
     UFUNCTION(BlueprintCallable)
     virtual void SetJoint();
 
+    //! Physics Constraints
     UPROPERTY(EditAnywhere, BlueprintReadWrite)
     UPhysicsConstraintComponent* Constraint = nullptr;
+
+    //! Smoothing TargetPose to #Constraint.
+    //! If this is false, step pose target are used by #SetPoseTarget
+    //! If this is true, pose target changes linearly with max vel in #TickComponent
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    bool bSmoothing = false;
+
+    //! Acceleration[cm/ss] used by velocity smoothing if #bVelocitySmoothing = true.
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    float LinearVelocitySmoothingAcc = 10;
+
+    //! Acceleration[deg/ss] used by velocity smoothing if #bVelocitySmoothing = true.
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    float AngularVelocitySmoothingAcc = 10;
+
+    //! [cm] tolerance for control
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    float PositionTolerance = 1;
+
+    //! [degree] tolerance for control
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    float OrientationTolerance = 1;
+
+    //! [cm/s] tolerance for control
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    float LinearVelocityTolerance = 10;
+    
+    //! [degree/s] tolerance for control
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    float AngularVelocityTolerance = 10;
+
+    //! [kg/ss]
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    float LinearSpring = 1000;
+
+    //! [kg/s]
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    float LinearBamper = 10;
+
+    //! [kg.cm/ss]
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    float LinearForceLimit = UE_BIG_NUMBER;
+
+    //! [kg.cm/ss]
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    float AngularSpring = 100000;
+
+    //! [kg.cm/s]
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    float AngularBamper = 10000;
+
+    //! [kg.deg.deg/ss]
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    float AngularForceLimit = UE_BIG_NUMBER;
+
+    //! if false, physics constraints parameters are set from #SetJoint .
+    //! Plese set true if you want to set physics constraints parameter manually.
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    bool bManualConstraintSetting = false;
+
+protected:
+
+    //! LinearVelocityTarget for smoothing
+    FVector MidLinearVelocityTarget = FVector::ZeroVector;
+
+    //! AngularVelocityTarget for smoothing
+    FVector MidAngularVelocityTarget = FVector::ZeroVector;
+
+    //! PositionTarget for smoothing
+    FVector MidPositionTarget = FVector::ZeroVector;
+
+    //! OrientationTarget for smoothing
+    FRotator MidOrientationTarget = FRotator::ZeroRotator;
```

Comment:
I used to experience non-reset member values between PIE->Stop->PIE->.. if without `UPROPERTY()` long ago in UE4, but might have been fixed by UE5 now